pennylane/math/multi_dispatch.py

```diff
@@ -532,6 +532,8 @@ def scatter_element_add(tensor, index, value, like=None):
     tensor([[ 0.1000, 20.2000,  0.3000],
             [ 0.4000,  0.5000, 10.6000]])
     """
+    if len(np.shape(tensor)) == 0 and index == ():
+        return tensor + value

```

Comment:
what is the consequence of this change?

Reply:
I must admit I forgot in which interface but there was a case where scatter-adding two 0-dimensional values crashed without this exception. I'll try to reproduce it.

Reply:
Indeed, without this patch
```python
qml.math.scatter_element_add(qml.math.array(0.5, like=interface), (), qml.math.array(0.2, like=interface))
```
fails for `interface="tensorflow"` with `IndexError: tuple index out of range` and for `interface="autograd"` (!) with `ValueError: At least one iterator operand must be non-NULL`

Reply:
ah! thanks for fixing this :)